src/context/orderReducer.tsx

```diff
@@ -24,63 +25,42 @@ export const orderReducer = (localOrder: LocalOrderDto, action: Action): LocalOr
   const { type, payload } = action;
 
   const productPayload = payload.product;
+  const lineItemQuantity = payload.quantity;
   const orderIdPayload = payload.orderId;
 
   const existingItem = localOrder.lineItems.find((item) => item.product.id === productPayload?.id);
   switch (type) {
     case ActionType.AddToLocalOrder:
-      if (!productPayload || existingItem) {
+      if (!productPayload || lineItemQuantity == undefined) {
         return localOrder;
       }
 
-      return {
-        ...localOrder,
-        lineItems: [...localOrder.lineItems, { product: productPayload, quantity: 1, subTotal: productPayload.price }],
-      };
-
-    case ActionType.IncreaseProductQuantity:
-      if (!productPayload) {
-        return localOrder;
-      }
-
-      return {
-        ...localOrder,
-        lineItems: [
-          ...localOrder.lineItems.map((item) => {
-            if (item.product.id === productPayload.id) {
-              const subtotal = item.product.price * item.quantity;
-              return { ...item, quantity: item.quantity++, subTotal: Math.round(subtotal * 1e2) / 1e2 };
-            }
-            return item;
-          }),
-        ],
-      };
-
-    case ActionType.DecreaseProductQuantityOrRemove:
-      if (!productPayload) {
-        return localOrder;
+      if (lineItemQuantity === 0) {
```

Comment:
Might be wrong because just had a glance. But is this necessary? Does this not return the same as the last return statement?